build.gradle.kts

```diff
@@ -7,3 +15,63 @@ repositories {
    google()
    gradlePluginPortal() // tvOS builds need to be able to fetch a kotlin gradle plugin
 }
+
+catalog {
+   val groupId = project.group.toString()
+   val extensionsGroupId = "$groupId.extensions"
+
+   println("using groupId: $groupId and version: $publishVersion for version catalog")
+
+   versionCatalog {
+      listOf(
+         "assertions-api",
+         "assertions-core",
+         "assertions-json",
+         "assertions-shared",
+         "assertions-sql",
+         "framework-api",
+         "framework-concurrency",
+         "framework-datatest",
+         "framework-discovery",
+         "framework-engine",
+         "property",
+         "runner-junit4",
+         "runner-junit5",
+         "extensions-htmlreporter", // Extension built and released with main kotest project
+         "extensions-junitxml" // Extension built and released with main kotest project
+      ).forEach { alias ->
+         library(alias, groupId, "kotest-$alias").version(publishVersion)
+      }
+
+      // Extensions
+      listOf(
+         "spring" to "1.1.2",
```

Comment:
How are we going to keep these versions up to date?

Reply:
Manually, I think. Up for suggestions though :)

If we publish with latest available version every time, then at least we should offer compatible versions. 

Given that extensions can release faster than Kotest itself, users might want to override the versions we specify here using [Gradle's mechanism for that
](https://docs.gradle.org/current/userguide/platforms.html#sec:overwriting-catalog-versions)

Reply:
I think no way we'll remember to do this manually.
Is there maybe a way to publish the version catalog outside the main pipeline? Otherwise as you said dependencies might get updated before we are able to update the version catalog

Reply:
Sure, it would be possible to externalize the version catalog pipeline. We could have a completely separate repo for the version catalog even. 

When would the version catalog be published, and how would the version of the catalog map to kotest version?

Reply:
@sksamuel what do you think?

Reply:
Is it possible to maybe update a catalog?

"Hey, for version 5.4.0 of the catalog, now take extension v1.2.1 instead of v1.2.0"

Then we could keep 'updating' the main catalog with the side versions.
No idea on the complexity of this, but knowing if this is possible at all would be nice

Reply:
I think that's forbidden, but I'm having trouble finding docs to confirm it. :(

Reply:
You can't override a release once it's on maven central

Reply:
But are version catalogues published to the maven central?

Reply:
I thought that they might be published to some gradle repository